common/trinary/flex_trit.c

```diff
@@ -190,10 +220,34 @@ size_t flex_trits_from_trytes(flex_trit_t *to_flex_trits, size_t to_len,
   memcpy(to_flex_trits, trytes, num_trytes);
 #elif defined(FLEX_TRIT_ENCODING_4_TRITS_PER_BYTE) || \
     defined(FLEX_TRIT_ENCODING_5_TRITS_PER_BYTE)
-  trit_t trits[to_len];
-  memset(trits, 0, to_len);
-  trytes_to_trits((tryte_t *)trytes, trits, num_trytes);
-  flex_trits_from_trits(to_flex_trits, to_len, trits, to_len, to_len);
+  union _shifter {
+    uint64_t val;
+    trit_t trits[8];
+  };
+  union _shifter shifter = {0};
+  size_t offset = 0;
+  size_t num_trits;
+  for (int i = 0, j = 0; num_trytes || offset; i++, j++) {
+    if (num_trytes) {
+      trytes_to_trits(&trytes[i], shifter.trits + offset, 1);
+      num_trytes -= 1;
+      offset += 3;
+      if (offset < NUM_TRITS_PER_FLEX_TRIT && num_trytes) {
+        i++;
+        trytes_to_trits(&trytes[i], shifter.trits + offset, 1);
+        num_trytes -= 1;
+        offset += 3;
+      }
+    }
+    num_trits = MIN(NUM_TRITS_PER_FLEX_TRIT, offset);
+    flex_trits_from_trits(to_flex_trits + j, num_trits, shifter.trits, num_trits, num_trits);
+#if BYTE_ORDER == LITTLE_ENDIAN
+    shifter.val = shifter.val >> (num_trits << 3);
```

Comment:
where is it being used?

Reply:
It is used to shift the bytes of `shifter.trits`